source/aspnetintegration.rst

```diff
@@ -263,3 +266,210 @@ A typical solution to this problem is to split up the class into multiple smalle
+        {
+            // Add application presentation components:
+            container.RegisterMvcControllers(app);
+            container.RegisterMvcViewComponents(app);
+
+            // Add application services.
+            //Register dependent services for the MyHostedService.  They will have to be singletons
+            //since hosted services are singletons in the ASP.NET Core configuration system.
+            container.Register<ISettingsObject, MySettingsObject>(Lifestyle.Singleton);
+            container.Register<ILogger, MyLogger>(Lifestyle.Singleton);
+
+            // Register MyHostedService in Simple Injector as Singleton
+            container.RegisterSingleton<MyHostedService>();
+
+            // Allow Simple Injector to resolve services from ASP.NET Core.
+            container.AutoCrossWireAspNetComponents(app);
+        }
+    }
+
+
+.. container:: Note
+
+    **NOTE**: The `documentation for hosted services <https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/hosted-services/>`_ says to register these services with the services.AddHostedService<T> extension method but that extension method doesn't give a way to provide either an instance or a factory.  Internally, the AddHostedService<T> extension method just calls the AddTransient<IHostedService, THostedService>() method. Later on, all of the IHostedService implementations are added as a dependency to HostedServiceExecuter which itself is added as a singleton. So, adding the hosted service implementations as singletons to the ASP.NET Core configuration system is an equivalent (and much clearer) way to add hosted services to the ASP.NET Core configuration system.
+
+The approach that will be laid out next is the preferred approach and it involves creating an adapter implementation that gets registered with the ASP.NET Core configuration system and forwards the calls to Simple Injector with the application-specific abstraction.
+
+The main idea with this approach is to have an application specific abstraction or abstractions which define the functionality to be performed in the IHostedService.  As an example, let's call this application-specific abstraction "IMyJob".  The implementation of IHostedService then becomes a simple adapter which can be used by ASP.NET Core to run application-specific dependencies in a hosted service similar to the way that the IControllerActivator abstraction implemented by SimpleInjectorControllerActivator provides an adapter for ASP.NET Core to be able to create Controllers with dependencies resolved by the Container.
+
+Given that, the IHostedService adapter might look like this:
+
+.. code-block:: c#
+
+    public class SimpleInjectorJobProcessorHostedService : IHostedService, IDisposable
+    {
+        private readonly Container _container;
+        private Timer _timer;
+
+        //Depending on how this adapter gets registered with the ASP.NET Core configuration system, this
```

Comment:
Your comment is 100% correct. Still, I would suggest removing it, as registration of the container is not required for the example below. So removing this comment, removes some noise.

Reply:
How about if I add a note after the example of the code block from the Startup class (before the line which begins with "This approach allows the application-specific code to remain..."):

    .. container:: Note

    **NOTE**: ConfigureServices will be called before all of the container registrations are finalized so the adapter and its registration must be written so that the container is not asked to resolve a registration prematurely.

Reply:
I understand what you're trying to teach the reader, but I still feel this is just more information than required and could actually still cause confusion.